go-controller/pkg/node/vrfmanager/vrf_manager.go

```diff
@@ -181,35 +202,26 @@ func (vrfm *Controller) sync(vrf vrf) error {
 			return fmt.Errorf("failed to get VRF device %s operationally up, err: %v", vrf.name, err)
 		}
 	}
-	existingEnslaves, err := getSlaveInterfaceNamesForVRF(vrfLink)
-	if err != nil {
-		return err
-	}
-	if existingEnslaves.Equal(vrf.enslavedInterfaces) {
-		return nil
-	}
-	for _, iface := range vrf.enslavedInterfaces.UnsortedList() {
-		if !existingEnslaves.Has(iface) {
-			err = enslaveInterfaceToVRF(vrf.name, iface)
-			if err != nil {
-				return fmt.Errorf("failed to enslave inteface %s into VRF device: %s, err: %v", iface, vrf.name, err)
-			}
+	if len(vrf.managedSlave) > 0 {
+		existingSlaves, err := getSlaveInterfaceNamesForVRF(vrfLink)
+		if err != nil {
+			return err
 		}
-	}
-	for _, iface := range existingEnslaves.UnsortedList() {
-		if !vrf.enslavedInterfaces.Has(iface) {
-			err = removeInterfaceFromVRF(vrf.name, iface)
-			if err != nil {
-				return fmt.Errorf("failed to remove inteface %s from VRF device: %s, err: %v", iface, vrf.name, err)
-			}
+		if existingSlaves.Has(vrf.managedSlave) {
+			return nil
```

Comment:
@trozet Theres a chance a VRF is configured on the system, ovn-k restarts, AddVRF is called, we come down to ln 211 and return because it already has the correct config, but we never added the VRF  to the cache in ln 219.

Reply:
good catch, fixing